### ISSUE TITLE
CLI error when adding S3 trigger

### ISSUE THREAD

**Issue by dmost714** [aws-amplify/amplify-cli#8858]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

v16.10.0

### Amplify CLI Version

6.4.0

### What operating system are you using?

Mac

### Amplify Categories

storage, function

### Amplify Commands

update

### Describe the bug

An error is generated attempting to create an S3 trigger via CLI. The error occurred when I initially created the new lambda trigger via the storage update wizard, as well as when I try to apply the existing lambda as a trigger. The lambda is visible in the console, but viewing 'Triggers' under the 'Configuration' tab in the AWS Console for the lambda reads 'No triggers' and there are no cloudwatch logs for the lambda generated when I add test files into the S3 bucket.

`amplify storage update`
? Please select from one of the below mentioned services: Content (Images, audio, video, etc.)
? Restrict access by? Individual Groups
? Select groups: GROUP1, GROUP2
? What kind of access do you want for GROUP1 users? create/update
? What kind of access do you want for GROUP2 users? read
? Do you want to add a Lambda Trigger for your S3 Bucket? Yes
✔ Select from the following options · Choose an existing function from the project
✔ Select from the following options · S3Trigger8d374d54
✅ Successfully updated resource S3Trigger8d374d54 locally
🛑 Cannot read properties of undefined (reading 'dependsOn')
✅ Successfully updated resource

### Expected behavior

The lambda should be triggered by S3 events in the associated bucket.

### Reproduction steps

1. amplify storage update
2. select Content (Images, audio, ...)
3. select "yes" when prompted to add a lambda trigger
4. create new lambda
5. amplify push
6. verify there are triggers setup for the deployed lambda
7. drop a file in the bucket
8. verify the cloudwatch events are generated

**Comment by josefaidt** [comment]
Hey @dmost714 :wave: thanks for raising this! I was able to reproduce this issue using Amplify CLI v6.4.0 as noted in your post, however I was able to update to the [latest version of the CLI (v7.3.1)](https://github.com/aws-amplify/amplify-cli/releases/tag/v7.3.1), migrate the resources, and successfully add a trigger to an s3 resource. Can you try upgrading and re-adding the trigger?

```
> amplify add storage

   ╭─────────────────────────────────────────────╮
   │                                             │
   │       Update available 6.4.0 → 7.3.1        │
   │   Run npm i -g @aws-amplify/cli to update   │
   │                                             │
   ╰─────────────────────────────────────────────╯

⚠️ Amplify command hooks are disabled in the current execution environment.
⚠️ See https://docs.amplify.aws/cli/usage/command-hooks/ for more information.
? Please select from one of the below mentioned services: Content (Images, audio, video, etc.)
✔ You need to add auth (Amazon Cognito) to your project in order to add storage for user files. Do you want to add auth now? (Y/n) · yes
Using service: Cognito, provided by: awscloudformation
 
 The current configured provider is Amazon Cognito. 
 
 Do you want to use the default authentication and security configuration? Default configuration
 Warning: you will not be able to edit these selections. 
 How do you want users to be able to sign in? Username
 Do you want to configure advanced settings? Yes, I want to make some additional changes.
 Warning: you will not be able to edit these selections. 
 What attributes are required for signing up? Email
 Do you want to enable any of the following capabilities? 
Successfully added auth resource 8858cd9f0072 locally

Some next steps:
"amplify push" will build all your local backend resources and provision it in the cloud
"amplify publish" will build all your local backend and frontend resources (if you have hosting category added) and provision it in the cloud

? Please provide a friendly name for your resource that will be used to label this category in the project: s32790a678
? Please provide bucket name: 885815116e8d4bd04d11b62a77ea90ccf804
? Who should have access: Auth and guest users
? What kind of access do you want for Authenticated users? create/update
? What kind of access do you want for Guest users? read
? Do you want to add a Lambda Trigger for your S3 Bucket? Yes
✔ Select from the following options · Create a new function
✅ Successfully added resource S3Trigger82d800be locally
? Do you want to edit the local S3Trigger82d800be lambda function now? No
⚠️ Auth configuration is required to allow unauthenticated users, but it is not configured properly.
Successfully updated auth resource locally.
✅ Successfully added resource s32790a678 locally

⚠️ If a user is part of a user pool group, run "amplify update storage" to enable IAM group policies for CRUD operations
✅ Some next steps:
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all of your local backend and front-end resources (if you added hosting category) and provisions them in the cloud


josef ~/D/p/a/r/8858
> amplify update storage
⚠️ Amplify command hooks are disabled in the current execution environment.
⚠️ See https://docs.amplify.aws/cli/usage/command-hooks/ for more information.
? Please select from one of the below mentioned services: Content (Images, audio, video, etc.)
? Who should have access: Auth and guest users
? What kind of access do you want for Authenticated users? create/update, read
? What kind of access do you want for Guest users? read
? Select from the following options Update the Trigger
✔ Select from the following options · Choose an existing function from the project
🛑 No functions were found in the project. Use 'amplify add function' to add a new function.
? Select from the following options Remove the trigger
✅ Successfully updated resource

> amplify update storage
⚠️ Amplify command hooks are disabled in the current execution environment.
⚠️ See https://docs.amplify.aws/cli/usage/command-hooks/ for more information.
? Please select from one of the below mentioned services: Content (Images, audio, video, etc.)
? Who should have access: Auth and guest users
? What kind of access do you want for Authenticated users? create/update, read
? What kind of access do you want for Guest users? read
? Do you want to add a Lambda Trigger for your S3 Bucket? Yes
✔ Select from the following options · Choose an existing function from the project
Only one option for [Select from the following options]. Selecting [S3Trigger82d800be].
✅ Successfully updated resource S3Trigger82d800be locally
🛑 Cannot read properties of undefined (reading 'dependsOn')
✅ Successfully updated resource
```

Upgrade to latest Amplify CLI with `npm i -g @aws-amplify/cli`

```
> amplify update storage
⚠️ Amplify command hooks are disabled in the current execution environment.
⚠️ See https://docs.amplify.aws/cli/usage/command-hooks/ for more information.
? Select from one of the below mentioned services: Content (Images, audio, video, etc.)
✔ A migration is needed to support latest updates on storage resources.
Recommended to try in a non-production environment first. Run "amplify env add" to create or clone an environment.
Learn more about this migration: https://docs.amplify.aws/cli/migration/override
Do you want to migrate storage resource "s32790a678"? (Y/n) · yes
✔ A migration is needed to support latest updates on auth resources.
Recommended to try in a non-production environment first. Run "amplify env add" to create or clone an environment.
Learn more about this migration: https://docs.amplify.aws/cli/migration/override
Do you want to migrate auth resource "8858cd9f0072"? (Y/n) · yes
✔ Who should have access: · Auth and guest users
✔ What kind of access do you want for Authenticated users? · create/update, read
✔ What kind of access do you want for Guest users? · read
✔ Do you want to add a Lambda Trigger for your S3 Bucket? (y/N) · yes
✔ Select from the following options · Choose an existing function from the project
Only one option for [Select from the following options]. Selecting [S3Trigger82d800be].
✅ Successfully updated resource
```

**Comment by dmost714** [comment]
Confirmed working in v7.3.3.
I am able to `amplify add storage` and add trigger lambda.
And the lambda gets called when files are added to the bucket.
